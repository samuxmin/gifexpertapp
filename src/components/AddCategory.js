import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState("")
    const handleInputChange = e =>{

        setInputValue(e.target.value)

        console.log('handleInputChange llamado')
    }

    const handleSubmit = e =>{
        e.preventDefault();

        if( inputValue.trim().length > 0){
        setCategories(categories=>[ inputValue, ...categories ])
        setInputValue("")
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="search">Search</label>
            <p>{inputValue}</p>
            <input 
                id="search"
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                 />
        </form>
    )
}

    AddCategory.propTypes = {
        setCategories: PropTypes.func.isRequired
    };

