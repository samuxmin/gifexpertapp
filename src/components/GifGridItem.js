import React from 'react'

export const GifGridItem = ( { id,title,url } ) => {

    return (
        <div className="card animate__bounceIn animate_animated">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
