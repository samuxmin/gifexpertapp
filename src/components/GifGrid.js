//import React, {useState, useEffect} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data:images, loading } = useFetchGifs(category);
     if(!loading && images.length === 0){
        return <>
        <h3 className="animate_animated animate__fadeInUp">{ category }</h3>
        <p>No se encontraron gifs para la categor√≠a: <i>{category}</i></p>
        </>
    }


    return (
        <>
            <h3 className="animate_animated animate__fadeInUp">{ category }</h3>

            {loading && <p className="animate_animated animate__flash">Loading...</p>}

            <div className="card-grid">
                    {
                    images.map(img => 
                    <GifGridItem 
                    key={img.id}
                    {...img}
                    />)
    }
        </div>
         </>
    )
}
