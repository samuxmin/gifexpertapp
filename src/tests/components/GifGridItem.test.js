import React from 'react';
import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from 'enzyme';


describe("pruebas en GifGridItem", () =>{
    
    const title = 'Un titulo';
    const url = 'https://localhost/algo.gif'
    
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should mostar gifgriditem correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    test('debe de tener un parrafo con el title', () =>{
        const p = wrapper.find('p');        
        expect(p.text().trim()).toBe(title) 
    })

    test('img debe tener un src y un titulo valido', () =>{
        const img = wrapper.find('img')
        expect( img.prop('src') ).toBe(url)
        expect( img.prop('alt') ).toBe(title)
    })

    test('div debe tener clase bounceIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        
        expect(className.includes('animate__bounceIn')).toBe(true)
    })
})