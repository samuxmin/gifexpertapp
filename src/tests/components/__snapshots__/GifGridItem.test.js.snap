// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem should mostar gifgriditem correctamente 1`] = `
<div
  className="card animate__bounceIn animate_animated"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.gif"
  />
  <p>
    Un titulo
  </p>
</div>
`;
